Objects/rangeobject.c

```diff
@@ -1135,13 +1145,19 @@ range_iter(PyObject *seq)
 
     /* If all three fields and the length convert to long, use the int
      * version */
-    lstart = PyLong_AsLong(r->start);
-    if (lstart == -1 && PyErr_Occurred()) {
+    lstop = PyLong_AsLong(r->stop);
+    if (lstop == -1 && PyErr_Occurred()) {
         PyErr_Clear();
         goto long_range;
     }
-    lstop = PyLong_AsLong(r->stop);
-    if (lstop == -1 && PyErr_Occurred()) {
+
+    if (r->start == _PyLong_GetZero() && r->step == _PyLong_GetOne() ) {
+        /* fast path for one argument range */
+        return fast_range_iter(0, lstop, 1, Py_MAX(0, (long)lstop));
```

Comment:
Why the type cast in `(long)lstop`?

Reply:
I suspect some leftover from old code, I removed the cast